proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+        case 'j':
+        case 'k':
+        case 'l':
+        case 'm':
+        case 'n':
+        case 'o':
+        case 'p':
+        case 'q':
+        case 'r':
+        case 's':
+        case 't':
+        case 'u':
+        case 'v':
+        case 'w':
+        case 'x':
+        case 'y':
+        case 'z':
+          append();
+          getChar();
+          ttype = PreprocessorParser.FILENAME;
+          break;
+        default:
+          break for_loop;
+      }
+    }
+    return makeToken(ttype);
+  }
+
+  ProToken id(int inputTokenType) throws IOException {
+    LOGGER.trace("Entering id()");
+
+    // Tokens that start with a-z or underscore
+    // - ID
+    // - FILENAME
+    // - keyword (testLiterals = true)
+    // Also inputTokenType can be ANNOTATION for a token that starts with '@'.
+    // Based on the PROGRESS online help, the following are the valid name characters.
+    // Undocumented: you can use a slash in an index name! Arg!
+    // Undocumented: the compiler allows you to start a block label with $
+    // If we find a back slash, we know we're into a filename.
+    // Extended characters (octal 200-377) can be used in identifiers, even at the beginning.
+    int ttype = inputTokenType;
+    for_loop : for (;;) {
+      switch (currChar) {
+        case '0':
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+        case 'a':
+        case 'b':
+        case 'c':
+        case 'd':
+        case 'e':
+        case 'f':
+        case 'g':
+        case 'h':
+        case 'i':
+        case 'j':
+        case 'k':
+        case 'l':
+        case 'm':
+        case 'n':
+        case 'o':
+        case 'p':
+        case 'q':
+        case 'r':
+        case 's':
+        case 't':
+        case 'u':
+        case 'v':
+        case 'w':
+        case 'x':
+        case 'y':
+        case 'z':
+        case '_':
+        case '-':
+        case '$':
+        case '#':
+        case '%':
+        case '&':
+        case '/':
+          // For tokens like ALT-* and CTRL-` :
+          // Emperically, I found that the following are the only other
+          // characters that can be put into a key label. Things
+          // like ALT-, must be put into quotes "ALT-,", because
+          // the comma has special meaning in 4gl code.
+          // Note also that extended characters can come after CTRL- or ALT-.
+          // ('!'|'@'|'^'|'*'|'+'|';'|'"'|'`')
+        case '!':
+        case '"':
+        case '*':
+        case '+':
+        case ';':
+        case '@':
+        case '^':
+        case '`':
+          append();
+          getChar();
+          break;
+        case '\\':
+        case '\'':
+          append();
+          getChar();
+          if (ttype == PreprocessorParser.ID)
+            ttype = PreprocessorParser.FILENAME;
+          break;
+        default:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1151)